packages/push/src/genome/plushy.rs

```diff
@@ -33,35 +36,45 @@ impl std::fmt::Debug for PushGene {
 }
 
 #[derive(Debug, Clone)]
-pub struct GeneGenerator {
+pub struct GeneGenerator<'a> {
     close_probability: f32,
-    instructions: Vec<PushInstruction>,
+    instructions: SliceCloning<'a, PushInstruction>,
```

Comment:
Maybe rename the field to `instruction_distribution`? The advantage of having a SliceCloning here is that that moves the checking for empty distributions out of the GeneGenerator. I would maybe be even more general here and do a GeneGenerator<T> where T: Distribution<PushInstruction> and then `instructions_distribution: T`, which would allow any distribution. Any thoughts?